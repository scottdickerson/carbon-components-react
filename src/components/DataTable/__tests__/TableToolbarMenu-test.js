/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import Download16 from '@carbon/icons-react/lib/download/16';
import { TableToolbarMenu } from '..';

describe('DataTable.TableToolbarMenu', () => {
  it('should render', () => {
    const wrapper = mount(
      <TableToolbarMenu
        className="custom-class"
        renderIcon={Download16}
        iconDescription="Add"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Custom icon in DataTable.TableToolbarMenu', () => {
  it('should render', () => {
    const iconAction = mount(
      <TableToolbarMenu renderIcon={Download16} iconDescription="Download" />
    );
    const originalIcon = mount(<Download16 />).find('svg');
    const icon = iconAction.find('svg');
    expect(icon.find(':not(svg):not(title)').html()).toBe(
      originalIcon.children().html()
    );
  });
});
