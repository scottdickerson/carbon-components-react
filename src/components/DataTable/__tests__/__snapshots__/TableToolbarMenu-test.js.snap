// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableToolbarMenu should render 1`] = `
<TableToolbarMenu
  className="custom-class"
  iconDescription="Add"
  renderIcon={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "height": 16,
        "preserveAspectRatio": "xMidYMid meet",
        "viewBox": "0 0 16 16",
        "width": 16,
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "displayName": "Download16",
      "propTypes": Object {
        "aria-hidden": [Function],
        "aria-label": [Function],
        "aria-labelledby": [Function],
        "children": [Function],
        "className": [Function],
        "height": [Function],
        "preserveAspectRatio": [Function],
        "tabIndex": [Function],
        "viewBox": [Function],
        "width": [Function],
        "xmlns": [Function],
      },
      "render": [Function],
    }
  }
>
  <ForwardRef(OverflowMenu)
    className="custom-class bx--toolbar-action bx--overflow-menu"
    flipped={true}
    renderIcon={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "height": 16,
          "preserveAspectRatio": "xMidYMid meet",
          "viewBox": "0 0 16 16",
          "width": 16,
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "displayName": "Download16",
        "propTypes": Object {
          "aria-hidden": [Function],
          "aria-label": [Function],
          "aria-labelledby": [Function],
          "children": [Function],
          "className": [Function],
          "height": [Function],
          "preserveAspectRatio": [Function],
          "tabIndex": [Function],
          "viewBox": [Function],
          "width": [Function],
          "xmlns": [Function],
        },
        "render": [Function],
      }
    }
    title="Add"
  >
    <OverflowMenu
      ariaLabel="list of options"
      className="custom-class bx--toolbar-action bx--overflow-menu"
      direction="bottom"
      flipped={true}
      iconDescription="open and close list of options"
      innerRef={null}
      menuOffset={[Function]}
      menuOffsetFlip={[Function]}
      onClick={[Function]}
      onClose={[Function]}
      onKeyDown={[Function]}
      onOpen={[Function]}
      open={false}
      renderIcon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "height": 16,
            "preserveAspectRatio": "xMidYMid meet",
            "viewBox": "0 0 16 16",
            "width": 16,
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "displayName": "Download16",
          "propTypes": Object {
            "aria-hidden": [Function],
            "aria-label": [Function],
            "aria-labelledby": [Function],
            "children": [Function],
            "className": [Function],
            "height": [Function],
            "preserveAspectRatio": [Function],
            "tabIndex": [Function],
            "viewBox": [Function],
            "width": [Function],
            "xmlns": [Function],
          },
          "render": [Function],
        }
      }
      tabIndex={0}
      title="Add"
    >
      <ClickListener
        onClickOutside={[Function]}
      >
        <div
          aria-expanded={false}
          aria-haspopup={true}
          aria-label="list of options"
          className="custom-class bx--toolbar-action bx--overflow-menu bx--overflow-menu"
          onClick={[Function]}
          onClose={[Function]}
          onKeyDown={[Function]}
          open={false}
          role="menu"
          tabIndex={0}
          title="Add"
        >
          <Download16
            aria-label="open and close list of options"
            className="bx--overflow-menu__icon"
            focusable="false"
            height={16}
            onClick={[Function]}
            onKeyDown={[Function]}
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <svg
              aria-label="open and close list of options"
              className="bx--overflow-menu__icon"
              focusable="false"
              height={16}
              onClick={[Function]}
              onKeyDown={[Function]}
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style={
                Object {
                  "willChange": "transform",
                }
              }
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                open and close list of options
              </title>
              <path
                d="M13 7l-.7-.7-3.8 3.8V1h-1v9.1L3.7 6.3 3 7l5 5zm0 5v2H3v-2H2v2c0 .6.4 1 1 1h10c.6 0 1-.4 1-1v-2h-1z"
              />
            </svg>
          </Download16>
        </div>
      </ClickListener>
    </OverflowMenu>
  </ForwardRef(OverflowMenu)>
</TableToolbarMenu>
`;
